contracts/access/AccessControl.sol

```diff
@@ -178,7 +169,9 @@ abstract contract AccessControl is Context, IAccessControl, ERC165 {
      * May emit a {RoleRevoked} event.
      */
     function renounceRole(bytes32 role, address account) public virtual override {
-        require(account == _msgSender(), "AccessControl: can only renounce roles for self");
+        if (account != _msgSender()) {
+            revert AccessControlIncorrectCaller();
```

Comment:
Not a fan of this error name. I don't think it's right to put the focus on the caller being incorrect. I think the goal of the function is to renounce the role from whoever the caller is, but the `account` argument serves as _confirmation_ that the caller is requesting this. I don't think the `account` argument name communicates that, so I would propose to rename it to `callerConfirmation` or `senderConfirmation`, and then call the error `BadConfirmation` or something to do with the confirmation argument.

Reply:
I'm also not a fan of this particular error. I discussed something similar with [@Amxx for Ownable2Step ownership acceptance](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4261#discussion_r1200279685) and we ended up agreeing that what you call _confirmation_ is a way of access, so we went for using `*UnauthorizedAccess`.

I just implemented the suggestion, but once you review Ownable2Step I'd ask: do you think we can have a similar naming convention between the two? Otherwise, you can solve this convo.

Reply:
IMO the error in Ownable2Step is not the same as this one so I wouldn't make them match.